lib/rgb_matrix/animation.ex

```diff
@@ -1,109 +1,88 @@
 defmodule RGBMatrix.Animation do
-  @moduledoc """
-  Provides a data structure and functions to define an RGBMatrix animation.
+  alias Layout.LED
 
-  There are currently two distinct ways to define an animation.
+  @callback new(leds :: list(LED.t()), config :: any) :: {render_in, any}
+  @callback render(state :: any, config :: any) ::
+              {list(RGBMatrix.any_color_model()), render_in, any}
+  @callback interact(state :: any, config :: any, led :: LED.t()) :: {render_in, any}
 
-  You may define an animation with a predefined `:frames` field. Each frame will advance every `:delay_ms` milliseconds.
-  These animations should use the `RGBMatrix.Animation.Static` `:type`. See the moduledocs of that module for
-  examples.
-
-  Alternatively, you may have a more dynamic animation which generates frames based on the current `:tick` of the
-  animation. See `RGBMatrix.Animation.{CycleAll, CycleLeftToRight, Pinwheel}` for examples.
-  """
-
-  alias __MODULE__
-  alias RGBMatrix.Frame
+  @type t :: %__MODULE__{
+          type: type,
+          config: any,
+          state: any
+        }
+  defstruct [:type, :config, :state]
```

Comment:
I added a similar comment about `config` yesterday. `state` may legitimately be `any` because it's up to the animation developer to decide if and how they want to use that property for their animation; we may not want to cause them trouble with dialyzer errors by over-specifying the type.

Reply:
I think I clarified this.